porch/controllers/packagevariantsets/pkg/controllers/packagevariantset/packagevariantset_controller.go

```diff
@@ -98,7 +111,20 @@ func (r *PackageVariantSetReconciler) Reconcile(ctx context.Context, req ctrl.Re
 	downstreams, err := r.unrollDownstreamTargets(ctx, pvs,
 		upstream.Package)
 	if err != nil {
-		return ctrl.Result{}, err
+		if strings.Contains(err.Error(), "unable to list specified objects in cluster") {
```

Comment:
I think it is primarily if the resource type is unknown that we should handle with conditions rather than retries. Can we do this by relying on `errors.As` rather than string check on our own error?

Reply:
I think the easiest way to check for this might be to look up the resource type in the RESTMapper as part of the validation step.

Reply:
I'm not sure how to use the RESTMapper, but I'm now using `meta.IsNoMatchError` to check the returned error type here. Happy to update it if you know of a better way to handle this with RESTMapper and can share some pointers.

Reply:
I think that looks like a good solution.